igcollect/statuscode2gaphite.py

```diff
@@ -0,0 +1,84 @@
+import argparse
+import datetime
+import time
+import socket
+
+from subprocess import run, PIPE
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Script to collect statuscodes from nginx')
+    parser.add_argument('-l', '--logfile',
+                        help='path to logfile')
+    parser.add_argument('-s', '--software', default='nginx',
+                        help='witch tool is using it (example: nginx)')
+    parser.add_argument('-a', '--addon', default='',
+                        help='additional awk conditions')
+    parser.add_argument('-x', '--position', type=int, default=9,
+                        help='were to search for the code in the log file '
+                             '(default= 9 nginx access.log)')
+    parser.add_argument('-t', '--time', type=int, default=1,
+                        choices=range(1, 5),
+                        help='the interval in (full)minutes the script runs.'
+                             'has to be the same time as the corresponding '
+                             'cronjob! '
+                             '(default= 1)')
+    parser.add_argument('-p', '--prefix',
+                        default='servers.{}.software.{}'.format(
+                            socket.gethostname().replace('.', '_'),
+                            parser.parse_args().software),
+                        help='the path to the value in Graphite '
+                             '(default= servers.(hostname).software.'
+                             '(software).status_codes)')
```

Comment:
You're duplicating defaults. It looks like argparse can handle this for you: https://docs.python.org/3/library/argparse.html#argparse.ArgumentDefaultsHelpFormatter